packages/vite-node/src/source-map.ts

```diff
@@ -63,7 +64,9 @@ export function withInlineSourcemap(
 
   // If the first line is not present on source maps, add simple 1:1 mapping ([0,0,0,0], [1,0,0,0])
   // so that debuggers can be set to break on first line
-  if (map.mappings.startsWith(';')) {
+  // Since Vite 6, import statements at the top of the file are preserved correctly,
+  // so we don't need to add this mapping anymore.
+  if (!options.noFirstLineMapping && map.mappings.startsWith(';')) {
```

Comment:
I added this check as per https://github.com/vitest-dev/vitest/pull/6732#issuecomment-2418803259.

Reply:
Wonder if we can guarantee that there's always something on the first line of the transpiled code. 🤔